aspnetcore/security/authentication/index.md

```diff
@@ -0,0 +1,107 @@
+
+Authentication middleware is added in `Startup.Configure` by calling the <xref:Microsoft.AspNetCore.Builder.AuthAppBuilderExtensions.UseAuthentication*> extension method on the app's `IApplicationBuilder`. This registers the  middleware which uses the previously registered authentication schemes. Call `UseAuthentication` before any middleware that depends on users being authenticated. When using endpoint routing, the call to `UseAuthentication` should go:
+
+* After `UseRouting`, so that route information is available for authentication decisions.
+* Before `UseEndpoints`, so that users are authenticated before accessing the endpoints.
+
+## Authentication Concepts
+
+### Authentication scheme
+
+An authentication scheme is:
+
+* A way of authenticating a user.
+* How the app responds if a user isn't authenticated.
+
+In ASP.NET Core, an authentication scheme consists of:
+
+* An authentication handler.
+* Options for configuring that specific instance of the handler.
+
+Authentication schemes have names so that they can be referenced in the app. For example, an authorization policy that specifies which authorization scheme (or schemes) should be used to authenticate the user. When configuring authentication, it's common to specify the default authentication scheme. The default scheme is used unless a resource requests a specific scheme. It's also possible to:
+
+* Specify different default schemes to use for authenticate, challenge, and forbid actions.
+* Combine multiple schemes into one using [policy schemes](xref:security/authentication/policyschemes).
+
+### Authentication handler
+
+An authentication handler:
+
+* Is a type that implements the behavior of a scheme.
+* Is derived from <xref:Microsoft.AspNetCore.Authentication.IAuthenticationHandler>.
+* Has the primary responsibility to authenticate users.
+
+Based on the authentication scheme's configuration and the incoming request context, authentication handlers:
+
+* Construct <xref:Microsoft.AspNetCore.Authentication.AuthenticationTicket> objects representing the user's identity if authentication is successful.
+* Return 'no result' or 'failure' if authentication is unsuccessful.
+* Have methods for challenge and forbid actions for when users attempt to access resources:
+  * They are unauthorized to access.
+  * When they are unauthenticated.
+
+### Challenge
+
+An authentication challenge is the action an authentication scheme takes when an unauthenticated user requests an endpoint that requires authentication. An authentication challenge is issued, for example, on a request to a restricted resource. Authentication challenge examples include:
+
+* Cookie authentication scheme redirecting the user to a login page.
+* The JWT bearer scheme returning a 401 result.
```

Comment:
and a `www-authenticate: bearer` header.

Reply:
Good point. I'll add this.